pkg/serviceimport/map.go

```diff
@@ -182,11 +184,22 @@ func (m *Map) Remove(serviceImport *mcsv1a1.ServiceImport) {
 		if len(remoteService.records) == 0 {
 			delete(m.svcMap, key)
 		} else if !remoteService.isHeadless {
-			remoteService.buildClusterInfoQueue()
+			remoteService.resetLoadBalancing()
 		}
 	}
 }
 
+func getServiceWeightFrom(annotation map[string]string) int64 {
+	if val, ok := annotation["load-balancer.submariner.io/weight"]; ok {
+		f, err := strconv.ParseInt(val, 0, 64)
+		if err != nil {
+			return f
+		}
```

Comment:
Sorry one more thing - we should at least log the error, eg 

`klog.Errorf("Error parsing the \"load-balancer.submariner.io/weight\" annotation from ServiceImport %q - using the default weight: %v, si.Name, err)`

Notice I included the `ServiceImport` name in the message for clarity. You can pass in the `ServiceImport`.